src/traits.rs

```diff
@@ -23,15 +23,13 @@ pub(crate) mod internal {
         fn maybe(fat: usize, capacity: Self::Field) -> Option<Self::NonZero>;
     }
 
-    /// Helper trait required by `Cow<T>` to extract capacity of owned
-    /// variant of `T`, and manage conversions.
-    ///
-    /// This can be only implemented on types that match requirements:
-    ///
-    /// + `T::Owned` has a `capacity`, which is an extra word that is absent in `T`.
-    /// + `T::Owned` with `capacity` of `0` does not allocate memory.
-    /// + `T::Owned` can be reconstructed from `*mut T` borrowed out of it, plus capacity.
-    pub unsafe trait Beef: ToOwned {
+    // for `~const Steak`
+    // I think that Steak (cooked beef) would contain "cooked" functions
+    // (computed at compile-time)
+    // However, `Vec::new()` also has const len/cap, but `std` does not implement it
```

Comment:
The thought so far has been to not put `const` on methods where the type isn't constructible `const` (or in `Vec`'s case, can only give one answer in `const` contexts).

I suspect this will relax in the future, but for the time being it allows greater impl flexibility especially w.r.t. future const allocation.